app/src/main/java/org/oppia/android/app/recyclerview/BindableAdapter.kt

```diff
@@ -150,7 +150,8 @@ class BindableAdapter<T : Any> internal constructor(
    * Instances of [SingleTypeBuilder] should be instantiated using [newBuilder].
    */
   class SingleTypeBuilder<T : Any>(
-    private val dataClassType: KClass<T>
+    private val dataClassType: KClass<T>,
+    private val fragment: Fragment
```

Comment:
Should we maybe pass this into ``BaseBuilder``'s constructor and remove ``setLifecycleOwner``? This allows us to leverage the primary benefit of being able to inject the fragment. Ditto for the multi-type below.

``BaseBuilder`` can also be private now, so we should change it to be private to minimize visibility.

Furthermore, both builder constructors can and should now be private since they're created via injectable factories.

Reply:
setLifecycleOwner shouldn't be needed anymore, which means BuilderType wouldn't be needed anymore.

One of the main goals of this refactor was to automatically attach the lifecycle owner in all cases since it's never wrong to attach, and it's easy to forget (which leads to hard-to-debug situations where LiveData simply doesn't work via databinding).

Reply:
Well noted Ben this makes sense.

Reply:
I think the only thing left for this comment is ensuring that the constructors become private for classes that don't need to be instantiated directly.

Reply:
Left a follow up comment regarding the constructor visibility.